src/main/java/school/hei/haapi/endpoint/rest/mapper/UserMapper.java

```diff
@@ -55,13 +60,15 @@ public Teacher toRestTeacher(User user) {
     teacher.setAddress(user.getAddress());
     teacher.setBirthPlace(user.getBirthPlace());
     teacher.setNic(user.getNic());
+    teacher.setProfilePicture(url);
 
     return teacher;
   }
 
   public Manager toRestManager(User user) {
     Manager manager = new Manager();
     manager.setId(user.getId());
+    String url = s3Service.getPresignedUrl(user.getRef(), 3600L);
```

Comment:
using ref as the key will output this :

bucket element KEY = STD21xxx

i do no think this is what you need.

Also, refactor your s3 usages and use the POJA's S3 bucketComponent

Reply:
in our case we use byte[] as file i should write a file to use the bucketComponent ?